Source/Endless_Hotel/Player/Controller/EHPlayerController.cpp

```diff
@@ -68,270 +59,272 @@ void AEHPlayerController::SetupInputComponent()
+	if (!SpringArm || !PlayerCameraComponent) return;
 
-    //SpringArm 카메라 위치 아래로 이동
-    if (SpringArm)
-        SpringArm->AddRelativeLocation(FVector(0.f, 0.f, -50.f));
+	if (!bIsFaceCovering) {
+		SpringArm->AddRelativeLocation(FVector(3.4f, 10.5f, 0.f));
+		bIsFaceCoverTransitioning = false;
+	}
 }
-void AEHPlayerController::Crouch_Completed()
+
+void AEHPlayerController::CheckForInteractables()
 {
-    if (!bIsCrouching)
-        return;
+	UCameraComponent* Camera = GetPlayerCamera();
+	if (!Camera) return;
+
+	FVector Start = Camera->GetComponentLocation();
+	FVector End = Start + Camera->GetForwardVector() * TraceDistance;
+
+	FHitResult HitResult;
+	FCollisionQueryParams Params;
+	Params.AddIgnoredActor(GetPawn());
 
-    bIsCrouching = false;
+	bool bHit = GetWorld()->LineTraceSingleByChannel(
+		HitResult,
+		Start,
+		End,
+		ECC_Visibility,
+		Params
+	);
 
-    GetCharacter()->UnCrouch();
-    GetCharacter()->GetCapsuleComponent()->SetCapsuleSize(45.f, 100.f);
+	// Debug line (시각 확인용)
+	// DrawDebugLine(GetWorld(), Start, End, FColor::Green, false, 0.0f, 0, 1.0f);
+
+	if (bHit && HitResult.GetActor())
+	{
+		AActor* HitActor = HitResult.GetActor();
+
+		if (HitActor->IsA(AInteract_Base::StaticClass()))
+		{
+			bCanInteract = true;
+			CurrentInteractActor = HitActor;
+			return;
+		}
+	}
 
-    //SpringArm 카메라 위치 위로 이동
-    if (SpringArm)
-        SpringArm->AddRelativeLocation(FVector(0.f, 0.f, 50.f));
+	// 트레이스 실패 or 유효하지 않음
+	bCanInteract = false;
+	CurrentInteractActor = nullptr;
 }
 
+void AEHPlayerController::OnInteract(const FInputActionValue& Value)
+{
+	if (!bCanInteract || !CurrentInteractActor) return;
 
-//FaceCover
-void AEHPlayerController::FaceCover_Triggered()
-{   
-    ACharacter* ControlledCharacter = Cast<ACharacter>(GetCharacter());
-    FVector fVelocity = ControlledCharacter->GetVelocity();
-    float Speed = fVelocity.Size();
-
-    if (Speed > 0.0f)
-        return; // 이동 중이면 전환 금지
-
-    // 상태 토글
-    bIsFaceCovering = !bIsFaceCovering;
-    bIsCameraFixed = !bIsCameraFixed;
-    bCanMove = !bCanMove;
-
-    // SpringArm & Camera 가져오기
-    if (!SpringArm)
-    {
-        SpringArm = ControlledCharacter->FindComponentByClass<USpringArmComponent>();
-    }
-
-    if (!PlayerCameraComponent)
-    {
-        PlayerCameraComponent = ControlledCharacter->FindComponentByClass<UCameraComponent>();
-    }
-
-    if (!SpringArm || !PlayerCameraComponent)
-        return;
-
-    // 카메라 전환 로직
-    if (bIsFaceCovering)
-    {
-        // 얼굴 보기 모드로 전환
-        SpringArm->AddRelativeLocation(FVector(-3.4f, -10.5f, 0.f));
-        PlayerCameraComponent->SetRelativeRotation(FRotator(0.f, -20.f, 0.f));
-    }
-    else
-    {
-        // 원래 위치로 복귀
-        SpringArm->AddRelativeLocation(FVector(3.4, 10.5f, 0.f));
-    }
+	if (AInteract_Base* Interactable = Cast<AInteract_Base>(CurrentInteractActor))
+	{
+		Interactable->Interact(this);
+	}
 }
 
+void AEHPlayerController::OnESCTriggered() {
```

Comment:
```suggestion
void AEHPlayerController::OnESCTriggered() {

}
```
해당 함수에 내용이 아무것도 없는데 왜 있어야 하는지 궁금합니다.

Reply:
게임 내에서 ESC 이벤트가 필요할 때를 위하여 작성해놓았습니다.